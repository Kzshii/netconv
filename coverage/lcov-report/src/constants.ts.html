
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for src/constants.ts</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> / <a href="index.html">src</a> constants.ts</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>1/1</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>0/0</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">100% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>1/1</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <pre><table class="coverage">
<tr><td class="line-count quiet"><a name='L1'></a><a href='#L1'>1</a>
<a name='L2'></a><a href='#L2'>2</a>
<a name='L3'></a><a href='#L3'>3</a>
<a name='L4'></a><a href='#L4'>4</a>
<a name='L5'></a><a href='#L5'>5</a>
<a name='L6'></a><a href='#L6'>6</a>
<a name='L7'></a><a href='#L7'>7</a>
<a name='L8'></a><a href='#L8'>8</a>
<a name='L9'></a><a href='#L9'>9</a>
<a name='L10'></a><a href='#L10'>10</a>
<a name='L11'></a><a href='#L11'>11</a>
<a name='L12'></a><a href='#L12'>12</a>
<a name='L13'></a><a href='#L13'>13</a>
<a name='L14'></a><a href='#L14'>14</a>
<a name='L15'></a><a href='#L15'>15</a>
<a name='L16'></a><a href='#L16'>16</a>
<a name='L17'></a><a href='#L17'>17</a>
<a name='L18'></a><a href='#L18'>18</a>
<a name='L19'></a><a href='#L19'>19</a>
<a name='L20'></a><a href='#L20'>20</a>
<a name='L21'></a><a href='#L21'>21</a>
<a name='L22'></a><a href='#L22'>22</a>
<a name='L23'></a><a href='#L23'>23</a>
<a name='L24'></a><a href='#L24'>24</a>
<a name='L25'></a><a href='#L25'>25</a>
<a name='L26'></a><a href='#L26'>26</a>
<a name='L27'></a><a href='#L27'>27</a>
<a name='L28'></a><a href='#L28'>28</a>
<a name='L29'></a><a href='#L29'>29</a>
<a name='L30'></a><a href='#L30'>30</a>
<a name='L31'></a><a href='#L31'>31</a>
<a name='L32'></a><a href='#L32'>32</a>
<a name='L33'></a><a href='#L33'>33</a>
<a name='L34'></a><a href='#L34'>34</a>
<a name='L35'></a><a href='#L35'>35</a>
<a name='L36'></a><a href='#L36'>36</a>
<a name='L37'></a><a href='#L37'>37</a>
<a name='L38'></a><a href='#L38'>38</a>
<a name='L39'></a><a href='#L39'>39</a>
<a name='L40'></a><a href='#L40'>40</a>
<a name='L41'></a><a href='#L41'>41</a>
<a name='L42'></a><a href='#L42'>42</a>
<a name='L43'></a><a href='#L43'>43</a>
<a name='L44'></a><a href='#L44'>44</a>
<a name='L45'></a><a href='#L45'>45</a>
<a name='L46'></a><a href='#L46'>46</a>
<a name='L47'></a><a href='#L47'>47</a>
<a name='L48'></a><a href='#L48'>48</a>
<a name='L49'></a><a href='#L49'>49</a>
<a name='L50'></a><a href='#L50'>50</a>
<a name='L51'></a><a href='#L51'>51</a>
<a name='L52'></a><a href='#L52'>52</a>
<a name='L53'></a><a href='#L53'>53</a>
<a name='L54'></a><a href='#L54'>54</a>
<a name='L55'></a><a href='#L55'>55</a>
<a name='L56'></a><a href='#L56'>56</a>
<a name='L57'></a><a href='#L57'>57</a>
<a name='L58'></a><a href='#L58'>58</a>
<a name='L59'></a><a href='#L59'>59</a>
<a name='L60'></a><a href='#L60'>60</a>
<a name='L61'></a><a href='#L61'>61</a>
<a name='L62'></a><a href='#L62'>62</a>
<a name='L63'></a><a href='#L63'>63</a>
<a name='L64'></a><a href='#L64'>64</a>
<a name='L65'></a><a href='#L65'>65</a>
<a name='L66'></a><a href='#L66'>66</a>
<a name='L67'></a><a href='#L67'>67</a>
<a name='L68'></a><a href='#L68'>68</a>
<a name='L69'></a><a href='#L69'>69</a>
<a name='L70'></a><a href='#L70'>70</a>
<a name='L71'></a><a href='#L71'>71</a>
<a name='L72'></a><a href='#L72'>72</a>
<a name='L73'></a><a href='#L73'>73</a>
<a name='L74'></a><a href='#L74'>74</a>
<a name='L75'></a><a href='#L75'>75</a>
<a name='L76'></a><a href='#L76'>76</a>
<a name='L77'></a><a href='#L77'>77</a>
<a name='L78'></a><a href='#L78'>78</a>
<a name='L79'></a><a href='#L79'>79</a>
<a name='L80'></a><a href='#L80'>80</a>
<a name='L81'></a><a href='#L81'>81</a>
<a name='L82'></a><a href='#L82'>82</a>
<a name='L83'></a><a href='#L83'>83</a>
<a name='L84'></a><a href='#L84'>84</a>
<a name='L85'></a><a href='#L85'>85</a>
<a name='L86'></a><a href='#L86'>86</a>
<a name='L87'></a><a href='#L87'>87</a>
<a name='L88'></a><a href='#L88'>88</a>
<a name='L89'></a><a href='#L89'>89</a>
<a name='L90'></a><a href='#L90'>90</a>
<a name='L91'></a><a href='#L91'>91</a>
<a name='L92'></a><a href='#L92'>92</a>
<a name='L93'></a><a href='#L93'>93</a>
<a name='L94'></a><a href='#L94'>94</a>
<a name='L95'></a><a href='#L95'>95</a>
<a name='L96'></a><a href='#L96'>96</a>
<a name='L97'></a><a href='#L97'>97</a>
<a name='L98'></a><a href='#L98'>98</a>
<a name='L99'></a><a href='#L99'>99</a>
<a name='L100'></a><a href='#L100'>100</a>
<a name='L101'></a><a href='#L101'>101</a>
<a name='L102'></a><a href='#L102'>102</a>
<a name='L103'></a><a href='#L103'>103</a>
<a name='L104'></a><a href='#L104'>104</a>
<a name='L105'></a><a href='#L105'>105</a>
<a name='L106'></a><a href='#L106'>106</a>
<a name='L107'></a><a href='#L107'>107</a></td><td class="line-coverage quiet"><span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export const Data = [
  {
    id: 0,
    name: 'Nintendo Switch',
    price: 1959.95,
    image:
      'https://cdn.pocket-lint.com/r/s/1200x630/assets/images/140007-games-review-nintendo-switch-review-image1-lp6zy9awm0.jpg',
  },
  {
    id: 1,
    name: 'Microsoft Xbox One S 1TB',
    price: 1599.95,
    image:
      'https://i.zst.com.br/images/console-xbox-one-s-1-tb-microsoft-hdr-4k-photo131312234-12-3d-14.jpg',
  },
  {
    id: 2,
    name: 'PlayStation 4 1TB',
    price: 2299.99,
    image:
      'https://media.playstation.com/is/image/SCEA/playstation-4-slim-vertical-product-shot-01-us-07sep16?$native_t$',
  },
  {
    id: 3,
    name: 'Controle Dualshock Midnight Blue - PS4',
    price: 242.99,
    image:
      'https://lojasaraiva.vteximg.com.br/arquivos/ids/12115195/1007196447.jpg?v=637142270587400000',
  },
  {
    id: 4,
    name: 'Jogo Fortnite - PS4',
    price: 150.99,
    image:
      'https://images-na.ssl-images-amazon.com/images/I/81JezUEQrAL._AC_SX425_.jpg',
  },
  {
    id: 5,
    name: 'New Nintendo 3DS XL',
    price: 1200.99,
    image:
      'https://images-na.ssl-images-amazon.com/images/I/71NPBdxtfsL._AC_SL1500_.jpg',
  },
  {
    id: 6,
    name: 'Donkey Kong Country Tropical Freeze - Nintendo Switch Mídia Digital',
    price: 159.99,
    image:
      'https://cdn-products.eneba.com/resized-products/cAgDmTm9r71F0ZJtrLWPyZXjgbHhjQhGEYugnefLyHE_390x400_1x-0.jpeg',
  },
  {
    id: 7,
    name: 'New Super Mario Bros. U Deluxe - Nintendo Switch',
    price: 189.9,
    image:
      'https://cdn-products.eneba.com/resized-products/ji562hhIgOe6q-P74LrAfFtfk_rG3cdAH36C7Aqcsvc_390x400_1x-0.jpeg',
  },
  {
    id: 8,
    name: 'Mario and Sonic nas Olimpiadas de Tóquio 2020',
    price: 229.9,
    image:
      'https://media.gamestop.com/i/gamestop/10178639/Mario-and-Sonic-at-the-Olympic-Games-Tokyo-2020?$pdp$',
  },
  {
    id: 9,
    name:
      'Gears Of War 5 Collectors Edition Jack Drone Remote Control + More - Xbox One',
    price: 1245.99,
    image: 'https://s3.amazonaws.com/comparegame/thumbnails/44027/large.jpg',
  },
  {
    id: 10,
    name: 'Gears of War: Ultimate Edition - Xbox One',
    price: 80.8,
    image:
      'https://a-static.mlcdn.com.br/618x463/gears-of-war-ultimate-edition-xbox-one-microsoft/bluewaves/6054551457/c92d65935dc99d8a725ed86161701496.jpg',
  },
  {
    id: 11,
    name: 'Pro Evolution Soccer eFootball PES 2020 - Playstation 4',
    price: 159.9,
    image:
      'https://images-na.ssl-images-amazon.com/images/I/71YFA3n13aL._AC_SX466_.jpg',
  },
  {
    id: 12,
    name: 'Pro Evolution Soccer eFootball PES 2020 - Playstation 4',
    price: 169.9,
    image: 'https://img.olx.com.br/images/50/501911087451251.jpg',
  },
  {
    id: 13,
    name: 'Tony Hawks Pro Skater 5 - PS4',
    price: 100.9,
    image:
      'https://http2.mlstatic.com/D_NQ_NP_881464-MLB41350448204_042020-W.jpg',
  },
  {
    id: 14,
    name: 'Skate 3 - Playstation 3',
    price: 89.9,
    image:
      'https://images-na.ssl-images-amazon.com/images/I/5197ha5%2BgqL._AC_SX385_.jpg',
  },
];
&nbsp;</pre></td></tr></table></pre>

                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Fri Apr 24 2020 00:24:12 GMT-0300 (GMT-03:00)
            </div>
        </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    