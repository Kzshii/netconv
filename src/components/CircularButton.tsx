import React, { FC } from 'react';
import { StyleSheet, View, Button } from 'react-native';

interface Props {
  onPress: () => void;
  title: string;
}

export const CircularButton: FC<Props> = ({ onPress, title }) => (
  <View style={styles.buttonContainer}>
    <Button color="#000" onPress={onPress} title={title} />
  </View>
);

const styles = StyleSheet.create({
  buttonContainer: {
    borderRadius: 26,
    paddingLeft: 5,
    paddingRight: 5,
    marginLeft: 10,
    marginRight: 10,
    borderWidth: 1,
    borderColor: '#bab5b5',
  },
});
