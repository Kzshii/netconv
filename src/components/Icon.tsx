import React, { FC } from 'react';
import {
  StyleSheet,
  Image,
  ImageProps,
  ImageSourcePropType,
} from 'react-native';

interface Props {
  path: ImageProps | ImageSourcePropType;
  style?: Record<string, string | number>;
}

export const Icon: FC<Props> = ({ path, style }) => (
  <Image style={{ ...styles.content, ...style }} source={path} />
);

const styles = StyleSheet.create({
  content: {
    width: 50,
    height: 50,
  },
});
