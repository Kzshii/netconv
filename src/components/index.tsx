export * from './CartIcon';
export * from './CartItem';
export * from './CircularButton';
export * from './DeleteButton';
export * from './Icon';
export * from './Item';
export * from './LogoIcon';
export * from './QuantityItem';
