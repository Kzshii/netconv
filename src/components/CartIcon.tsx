import React, { FC } from 'react';
import { useSelector } from 'react-redux';
import { productsSelector } from '../store/selectors';
import { StyleSheet, View, Text, TouchableHighlight } from 'react-native';
import { Icon } from './Icon';
import { useNavigation } from '@react-navigation/native';

export const CartIcon: FC = () => {
  const products = useSelector(productsSelector);
  const navigation = useNavigation();
  return (
    <TouchableHighlight
      activeOpacity={0.6}
      underlayColor="transparent"
      onPress={() => navigation.navigate('Cart')}
    >
      <View style={styles.cart}>
        <Icon
          style={styles.cartIcon}
          path={require('../../assets/images/cart.png')}
        />
        <View style={styles.cartContent}>
          <Text style={styles.cartCount}>{products.length}</Text>
        </View>
      </View>
    </TouchableHighlight>
  );
};

const styles = StyleSheet.create({
  cart: {
    alignItems: 'center',
    flexDirection: 'row',
    right: 18,
  },
  cartIcon: {
    width: 30,
    height: 30,
    right: 12,
  },
  cartContent: {
    position: 'absolute',
    top: -8,
    right: 0,
    backgroundColor: '#fff',
    borderRadius: 12,
    paddingLeft: 5,
    paddingRight: 5,
  },
  cartCount: {
    right: 0,
    fontSize: 16,
    fontWeight: 'bold',
  },
});
