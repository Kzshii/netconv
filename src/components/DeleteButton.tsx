import React, { FC } from 'react';
import { StyleSheet, View } from 'react-native';
import { TouchableHighlight } from 'react-native';
import { Icon } from './Icon';
import { ProductsActions } from '../store/actions/products';
import { useDispatch } from 'react-redux';
import { Product } from '../types';

interface Props {
  product: Product;
}

export const DeleteButton: FC<Props> = ({ product }) => {
  const dispatch = useDispatch();
  const onDelete = () => {
    dispatch(ProductsActions.deleteProduct(product));
  };
  return (
    <TouchableHighlight
      activeOpacity={0.6}
      underlayColor="transparent"
      onPress={onDelete}
    >
      <View style={styles.container}>
        <Icon
          style={{ width: 20, height: 20 }}
          path={require('./../../assets/images/trash.png')}
        />
      </View>
    </TouchableHighlight>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 12,
  },
  text: {
    color: '#fff',
  },
  quantityContainer: {
    alignItems: 'center',
  },
  title: {
    fontSize: 22,
    maxWidth: 150,
  },
  price: {
    marginTop: 12,
    fontSize: 16,
    fontWeight: 'bold',
  },
  image: {
    borderRadius: 12,
    width: 100,
    height: 100,
    marginRight: 15,
  },
});
