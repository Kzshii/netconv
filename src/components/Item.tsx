import React, { FC } from 'react';
import { StyleSheet, View, Text, TouchableHighlight } from 'react-native';
import { Icon } from './Icon';
import { Product } from '../types';

interface Props {
  product: Product;
  onPress: (item: Product) => void;
}

export const Item: FC<Props> = ({ product, onPress }) => (
  <TouchableHighlight
    style={styles.touchable}
    activeOpacity={0.6}
    underlayColor="transparent"
    onPress={() => onPress(product)}
  >
    <View style={styles.container}>
      <View style={styles.descriptionContent}>
        <Icon style={styles.image} path={{ uri: product.image }} />
        <Text numberOfLines={2} style={styles.title}>
          {product.name}
        </Text>
      </View>
      <Text style={styles.price}>
        R$ {product.price.toLocaleString('br-BR')}
      </Text>
    </View>
  </TouchableHighlight>
);

const styles = StyleSheet.create({
  container: {
    padding: 12,
    justifyContent: 'center',
    alignItems: 'center',
  },
  descriptionContent: {
    justifyContent: 'center',
    alignItems: 'center',
  },
  title: {
    fontSize: 14,
    maxWidth: 150,
    marginBottom: 3,
  },
  price: {
    fontSize: 16,
    fontWeight: 'bold',
  },
  image: {
    borderWidth: 1,
    borderColor: '#9ea0a3',
    borderRadius: 12,
    width: 100,
    height: 100,
    marginBottom: 5,
  },
  touchable: {
    flexBasis: 0,
    flexGrow: 1,
  },
});
