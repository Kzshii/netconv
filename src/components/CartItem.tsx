import React, { FC } from 'react';
import { StyleSheet, View, Text, TouchableHighlight } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { Product } from '../types';
import { QuantityItem } from './QuantityItem';
import { Icon } from './Icon';
import { DeleteButton } from './DeleteButton';

interface Props {
  product: Product;
  setItem: (product: Product) => void;
}

export const CartItem: FC<Props> = ({ product, setItem }) => {
  const navigation = useNavigation();
  const onItemClick = (product: Product) => {
    navigation.navigate('Description', { product });
  };
  return (
    <View style={styles.container}>
      <TouchableHighlight
        activeOpacity={0.6}
        underlayColor="transparent"
        onPress={() => onItemClick(product)}
      >
        <View style={styles.descriptionContainer}>
          <Icon style={styles.image} path={{ uri: product.image }} />
          <Text numberOfLines={2} style={styles.title}>
            {product.name}
          </Text>
        </View>
      </TouchableHighlight>
      <View style={styles.descriptionContainer}>
        <DeleteButton product={product} />
        <View style={styles.quantityContainer}>
          <QuantityItem setItem={setItem} product={product} />
          <Text style={styles.price}>
            R$ {product.price.toLocaleString('br-BR')}
          </Text>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    padding: 12,
    justifyContent: 'space-between',
    alignItems: 'center',
    flexDirection: 'row',
  },
  descriptionContainer: {
    alignItems: 'center',
    flexDirection: 'row',
  },
  quantityContainer: {
    alignItems: 'center',
  },
  title: {
    fontSize: 14,
    maxWidth: 150,
  },
  price: {
    marginTop: 12,
    fontSize: 16,
    fontWeight: 'bold',
  },
  image: {
    borderWidth: 1,
    borderColor: '#9ea0a3',
    borderRadius: 12,
    width: 60,
    height: 60,
    marginRight: 15,
  },
});
