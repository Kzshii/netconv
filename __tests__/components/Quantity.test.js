import 'react-native';
import React from 'react';
import { QuantityItem, CircularButton } from '../../src/components';
import { shallow } from 'enzyme';

describe('<QuantityItem />', () => {
  const product = {
    id: 1,
    name: 'Test',
    price: 1.99,
    image: 'https://reactnative.dev/img/tiny_logo.png',
    quantity: 1
  };

  test('renders correctly QuantityItem', () => {
    const wrapper = shallow(
      <QuantityItem product={product} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('should increment', () => {
    const setItem = jest.fn();
    const wrapper = shallow(
      <QuantityItem product={product} setItem={setItem}/>
    );
    wrapper.find(CircularButton).last().props().onPress();
    expect(wrapper).toMatchSnapshot();
  })

  test('should decrement', () => {
    const setItem = jest.fn();
    const wrapper = shallow(
      <QuantityItem product={product} setItem={setItem}/>
    );
    wrapper.find(CircularButton).first().props().onPress();
    expect(wrapper).toMatchSnapshot();
  })
});
