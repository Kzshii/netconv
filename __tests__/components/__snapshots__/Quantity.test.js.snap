// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<QuantityItem /> renders correctly QuantityItem 1`] = `ShallowWrapper {}`;

exports[`<QuantityItem /> should decrement 1`] = `ShallowWrapper {}`;

exports[`<QuantityItem /> should increment 1`] = `ShallowWrapper {}`;
